sdk/keyvault/azure-security-test-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/KeyVaultCertificatesTest.java

```diff
@@ -0,0 +1,116 @@
+        + "KEt94z88MPX18h8ud14uRVmUCYVZrZeswdE2tO1BpazrXELHuXCtrjGxsDDjDzeP"
+        + "98aFI9kblkqoJS4TsmloLEjwZLm80cyJDEmpXXMtR7C0FFXFI1BAtIa4mxSgBLsT"
+        + "L4GVPEGNANR8COYkHQIDAQABo4GjMIGgMA4GA1UdDwEB/wQEAwIFoDAJBgNVHRME"
+        + "AjAAMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAkBgNVHREEHTAbghlo"
+        + "dW5kcmVkLXllYXJzLmV4YW1wbGUuY29tMB8GA1UdIwQYMBaAFOGTt4H3ho30O4e+"
+        + "hebwJjm2VMvIMB0GA1UdDgQWBBThk7eB94aN9DuHvoXm8CY5tlTLyDANBgkqhkiG"
+        + "9w0BAQsFAAOCAQEAGp8mCioVCmM+kZv6r+K2j2uog1k4HBwN1NfRoSsibDB8+QXF"
+        + "bmNf3M0imiuR/KJgODyuROwaa/AalxNFMOP8XTL2YmP7XsddBs9ONHHQXKjY/Ojl"
+        + "PsIPR7vZjwYPfEB+XEKl2fOIxDQQ921POBV7M6DdTC49T5X+FsLR1AIIfinVetT9"
+        + "QmNuvzulBX0T0rea/qpcPK4HTj7ToyImOaf8sXRv2s2ODLUrKWu5hhTNH2l6RIkQ"
+        + "U/aIAdQRfDaSE9jhtcVu5d5kCgBs7nz5AzeCisDPo5zIt4Mxej3iVaAJ79oEbHOE"
+        + "p192KLXLV/pscA4Wgb+PJ8AAEa5B6xq8p9JO+Q==";
+
+    @BeforeAll
+    public static void setEnvironmentProperty() {
+        PropertyConvertorUtils.putEnvironmentPropertyToSystemProperty(
+            Arrays.asList("AZURE_KEYVAULT_URI",
+                "AZURE_KEYVAULT_TENANT_ID",
+                "AZURE_KEYVAULT_CLIENT_ID",
+                "AZURE_KEYVAULT_CLIENT_SECRET")
+        );
+        Security.insertProviderAt(new KeyVaultJcaProvider(), 1);
+        certificateName = System.getenv("AZURE_KEYVAULT_CERTIFICATE_NAME");
+    }
+
+    @Test
+    public void testGetKeyStore() throws Exception {
+        KeyStore keyStore = PropertyConvertorUtils.getKeyVaultKeyStore();
+        assertNotNull(keyStore.getCertificate(certificateName));
+        assertTrue(keyStore.containsAlias(certificateName));
+        X509Certificate certificate = getTestCertificate();
+
+        keyStore.setCertificateEntry("setcert", certificate);
+        assertNotNull(keyStore.getCertificateAlias(certificate), "setcert");
+    }
+
+    private X509Certificate getTestCertificate() {
+        X509Certificate certificate;
+
+        try {
+            byte[] certificateBytes = Base64.getDecoder().decode(TEST_CERTIFICATE);
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
+            certificate = (X509Certificate) cf.generateCertificate(new ByteArrayInputStream(certificateBytes));
+        } catch (CertificateException e) {
+            throw new ProviderException(e);
+        }
+        return certificate;
+    }
+
+
+
+    @Test
+    public void testCertificatesRefreshInterval() throws Exception {
+        System.setProperty("azure.keyvault.jca.certificates-refresh-interval", "15000");
+        KeyStore keyStore = PropertyConvertorUtils.getKeyVaultKeyStore();
+        X509Certificate certificate = getTestCertificate();
+
+        assertNotNull(keyStore.getCertificate(certificateName));
+        keyStore.deleteEntry(certificateName);
+        keyStore.setCertificateEntry(certificateName, certificate);
+        Thread.sleep(30000);
+        assertNull(keyStore.getCertificateAlias(certificate));
```

Comment:
When the certificate of `0myalias` exists in both `keyVaultCertificate` and `ClasspathCertificate`, the certificate in `keyVaultCertificate` should really work.
When searching for an alias based on a certificate, the first step is to take out all available aliases in the `keyVaultCertificate` and `ClasspathCertificate`. Certificates with same alias should only appear once.
Since the authority of `keyVaultCertificate` is higher than `ClasspathCertificate` ([my opinion](https://github.com/Azure/azure-sdk-for-java/pull/20765#discussion_r635682234)), the certificate in `ClasspathCertificate` will be ignored.